src/core/vector/qgsvectorlayertemporalproperties.cpp

```diff
@@ -514,37 +512,31 @@ QString QgsVectorLayerTemporalProperties::createFilterString( const QgsVectorLay
         case QgsUnitTypes::TemporalIrregularStep:
           return QString();
       }
-      return QStringLiteral( "(%1 %2 %3 OR %1 IS NULL) AND ((%1 + %4 %5 %6) OR %7 IS NULL)" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-             range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
```

Comment:
we need to leave the `range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),`  logic here -- see above